drivers/netdev_ieee802154_submac/netdev_ieee802154_submac.c

```diff
@@ -33,23 +32,12 @@ static void _ack_timeout(void *arg)
 
 static netopt_state_t _get_submac_state(ieee802154_submac_t *submac)
 {
-    ieee802154_submac_state_t state = ieee802154_get_state(submac);
-
-    netopt_state_t netopt_state;
-    switch (state) {
-        case IEEE802154_STATE_OFF:
-            netopt_state = NETOPT_STATE_SLEEP;
-            break;
-        case IEEE802154_STATE_IDLE:
-            netopt_state = NETOPT_STATE_STANDBY;
-            break;
-        case IEEE802154_STATE_LISTEN:
-        default:
-            netopt_state = NETOPT_STATE_IDLE;
-            break;
+    if (ieee802154_submac_state_is_idle(submac)) {
+        return NETOPT_STATE_SLEEP;
+    }
+    else {
+        return NETOPT_STATE_IDLE;
     }
```

Comment:
Since this branch of `netdev_ieee802154` is in top of the `submac`, `NETOPT_STATE_STANDBY` makes not so much sense here. 

The idea is that the SubMAC provides the lowest power consumption for a given operation mode. Therefore, `NETOPT_STATE_IDLE` should be receiving and `NETOPT_STATE_SLEEP` should put the device in its lowest consumption profile. In fact, we should probably try to keep the radio sleeping as much as possible:  since the SubMAC handles all critical timings (ACKs, CSMA-CA, retransmissions), going to the lowest consumption mode won't add timing issues due to long wake-up times.

This is not implemented yet, but the dea is that the SubMAC is able to re-configure the radio if it goes to deep sleep (we added a REG_RETENTION cap).